torch/distributed/rpc/api.py

```diff
@@ -592,9 +594,15 @@ def remote(to, func, args=None, kwargs=None, timeout=UNSET_RPC_TIMEOUT):
                 **kwargs,
             )
         else:
-            (pickled_python_udf, tensors) = _default_pickler.serialize(
-                PythonUDF(func, args, kwargs)
-            )
+            arg_cache_key = (func, args, str(kwargs))
```

Comment:
`str(kwargs)` might not work correctly since we get different results if the order of the kwargs are different:

```
>>> dict1 = {"a":1, "b":2}
>>> dict2 = {"b":2, "a":1}
>>> str(dict1)
"{'a': 1, 'b': 2}"
>>> str(dict2)
"{'b': 2, 'a': 1}"
```

Reply:
I think the dictionary arguments will likely be consistent across most invocations (and dict keys are guaranteed ordered after python 3.5 I believe), but this could happen in the case where the kwargs is constructed slightly differently across successive calls. Usually we expect the use case to look like as follows, where this would not be an issue:
```
for trainer in trainers:
    rref = rpc.remote(trainer, func, args=(t1, t2), kwargs={"foo": 1, "bar": 1})
```

Although if we think this assumption will miss many issues, we can probably cache `str(OrderedDict(kwargs.items()))` performance permitting which would keep the equality